java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/EventHubsKafkaConsumerController.java

```diff
@@ -0,0 +1,144 @@
+import static io.lenses.java.streamreactor.connect.azure.eventhubs.mapping.SourceRecordMapper.mapSourceRecordWithoutHeaders;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.GROUP_ID_CONFIG;
+
+import io.lenses.java.streamreactor.connect.azure.eventhubs.config.AzureEventHubsConfig;
+import io.lenses.java.streamreactor.connect.azure.eventhubs.config.AzureEventHubsConfigConstants;
+import io.lenses.java.streamreactor.connect.azure.eventhubs.source.TopicPartitionOffsetProvider.AzureOffsetMarker;
+import io.lenses.java.streamreactor.connect.azure.eventhubs.source.TopicPartitionOffsetProvider.AzureTopicPartitionKey;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.common.config.SaslConfigs;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.source.SourceRecord;
+
+/**
+ * Class is a bridge between EventHub KafkaConsumers and AzureEventHubsSourceTask. It verifies the configuration
+ * of kafka consumers and instantiates them, then allows AzureEventHubsSourceTask to pull for SourceRecords.
+ */
+@Slf4j
+public class EventHubsKafkaConsumerController {
+
+  private final AzureEventHubsConfig azureEventHubsConfig;
+  private final KafkaConsumerProvider kafkaConsumerProvider;
+  private final BlockingQueue<ConsumerRecords<String, String>> recordsQueue;
+  private final BlockingQueuedKafkaConsumer queuedKafkaConsumer;
+
+  /**
+   * Constructs EventHubsKafkaConsumerController.
+   *
+   * @param azureEventHubsConfig config for SourceTask
+   * @param kafkaConsumerProvider provider of KafkaConsumers
+   * @param recordsQueue queue that contains EventHub records.
+   */
+  public EventHubsKafkaConsumerController(AzureEventHubsConfig azureEventHubsConfig,
+      KafkaConsumerProvider kafkaConsumerProvider,
+      BlockingQueue<ConsumerRecords<String, String>> recordsQueue) {
+    this.azureEventHubsConfig = azureEventHubsConfig;
+    this.kafkaConsumerProvider = kafkaConsumerProvider;
+    this.recordsQueue = recordsQueue;
+
+    queuedKafkaConsumer = createQueuedConsumer();
+  }
+
+  private BlockingQueuedKafkaConsumer createQueuedConsumer() {
+    Properties consumerProperties = extractConsumerProperties();
+    return kafkaConsumerProvider.createConsumer(consumerProperties, recordsQueue);
+  }
+
+  private Properties extractConsumerProperties() {
```

Comment:
Why not extract the Kafka Consumer client initialisation?

Reply:
yeah, we may want to just pass it to `KafkaConsumerProvider` depending on what you mean [here](https://github.com/lensesio/stream-reactor/pull/1033#discussion_r1508852691)

But this would delegate properties extraction which would be nice.

Reply:
This was delegated to `BlockingQueueProducerProvider`